packages/es-components/src/components/containers/popoverTest/PopoverTest.js

```diff
@@ -0,0 +1,273 @@
+    this.closeBtn = findDOMNode(this.closeBtnRef);
+    this.triggerBtn = findDOMNode(this.triggerBtnRef);
+  }
+
+  toggleShow = () => {
+    const { isOpen } = this.state;
+
+    setTimeout(() => {
+      const focusableContent = this.content.querySelector('a, button');
+      if (focusableContent) {
+        focusableContent.focus();
+      } else {
+        this.header.focus();
+      }
+    }, 200);
+
+    if (this.closeBtn) {
+      this.triggerBtn.focus();
+    }
+
+    this.setState({ isOpen: !isOpen });
+  };
+
+  hidePopover = () => {
+    if (this.state.isOpen) {
+      setTimeout(() => {
+        this.triggerBtn.focus();
+      }, 150);
+    }
+    this.setState({ isOpen: false });
+  };
+
+  render() {
+    const {
+      name,
+      title,
+      content,
+      placement,
+      children,
+      arrowSize,
+      hasCloseButton,
+      hasAltCloseButton,
+      suppressUnderline,
+      isLinkButton,
+      ariaLabel,
+      theme
+    } = this.props;
+
+    const hasTitle = title !== undefined;
+    const hasAltCloseWithNoTitle = !hasTitle && hasAltCloseButton;
+    const showCloseButton = hasCloseButton && !hasAltCloseButton;
+
+    const triggerButton = (
+      <TriggerButton
+        isLinkButton={isLinkButton}
+        handleOnClick={this.toggleShow}
+        suppressUnderline={suppressUnderline}
+        ref={btn => {
+          this.triggerBtnRef = btn;
+        }}
+        aria-expanded={this.state.isOpen}
+      >
+        {children}
+        <TriggerButtonLabel>{ariaLabel}</TriggerButtonLabel>
+      </TriggerButton>
+    );
+
+    const closeButton = (
+      <Button
+        handleOnClick={this.toggleShow}
+        ref={btn => {
+          this.closeBtnRef = btn;
+        }}
+      >
+        Close
+      </Button>
+    );
+
+    const altCloseButton = (
+      <AlternateCloseButton
+        aria-label="Close"
+        hasTitle={hasTitle}
+        handleOnClick={this.toggleShow}
+        ref={btn => {
+          this.closeBtnRef = btn;
+        }}
+      >
+        <Icon name="remove" />
+      </AlternateCloseButton>
+    );
+
+    return (
+      <ThemeProvider theme={theme}>
+        <Container>
+          <Popup
+            name={name}
+            trigger={triggerButton}
+            placement={placement}
+            transitionIn={this.state.isOpen}
+            onHide={this.hidePopover}
+            arrowSize={arrowSize}
+            hasTitle={hasTitle}
+            theme={theme}
+          >
+            <PopoverContainer
+              role="dialog"
+              ref={elem => {
+                this.contentRef = elem;
+              }}
+              tabIndex={-1}
+            >
+              <PopoverHeader
+                hasTitle={hasTitle}
+                ref={elem => {
+                  this.headerRef = elem;
+                }}
+                tabIndex={-1}
+              >
+                {hasTitle ? <TitleBar>{title}</TitleBar> : ''}
+                {hasAltCloseButton ? altCloseButton : ''}
```

Comment:
Do these need to be ternaries or can we simplify them down to `&&` operators?

```js
{ hasTitle && <TitleBar>{title}</TitleBar> }
{ hasAltCloseButton && altCloseButton }
```

Reply:
What does this give us that the ternaries do not? Personally I prefer the ternaries because I think it's easier to understand what they're doing over the `&&` operators, since the `&&` operators will essentially always have one true value.

Reply:
I think using the `&&` operator looks a lot cleaner and prefer that over a ternary when it has a null or empty string case.

Reply:
The ternaries are overkill in these instances. We don't have any alternatives to show if either`hasTitle` or `hasAltCloseButton` is false. The empty strings aren't providing any more value over the [`&&` operator](https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator). If we were showing something more meaningful than empty strings I'd be fine with keeping them, otherwise `&&` is a cleaner and concise approach.